tests/test_scatterPlot.py

```diff
@@ -0,0 +1,131 @@
+filename_figure_ref = os.path.join(ROOT, "data", "test_scatterPlot.png")
+
+
+class ScatterPlotWithTwoHistsTaskTestCase(lsst.utils.tests.TestCase):
+    """ScatterPlotWithTwoHistsTask test case."""
+    def setUp(self):
+        self.testDir = tempfile.mkdtemp(dir=ROOT, prefix="test_output")
+
+        # Set up a quasi-plausible measurement catalog
+        mag = 12.5 + 2.5*np.log10(np.arange(10, 100000))
+        flux = 10**(-0.4*(mag - (mag[-1] + 1)))
+        rng = default_rng(0)
+        extendedness = 0. + (rng.uniform(size=len(mag)) < 0.99*(mag - mag[0])/(mag[-1] - mag[0]))
+        flux_meas = flux + rng.normal(scale=np.sqrt(flux*(1 + extendedness)))
+        flux_err = np.sqrt(flux_meas * (1 + extendedness))
+        good = (flux_meas/np.sqrt(flux * (1 + extendedness))) > 3
+        extendedness = extendedness[good]
+        flux = flux[good]
+        flux_meas = flux_meas[good]
+        flux_err = flux_err[good]
+
+        # Configure the plot to show observed vs true mags
+        config = ScatterPlotWithTwoHistsTaskConfig(
+            axisLabels={'x': 'mag', 'y': 'mag meas - ref', 'mag': 'mag'},
+        )
+        config.selectorActions.flagSelector.bands = ['i']
+        config.axisActions.yAction = MagDiff(col1='refcat_flux', col2='refcat_flux')
+        config.nonBandColumnPrefixes.append('refcat')
+        config.sourceSelectorActions.galaxySelector = GalaxyIdentifier
+        config.highSnStatisticSelectorActions.statSelector.threshold = 50
+        config.lowSnStatisticSelectorActions.statSelector.threshold = 20
+        self.task = ScatterPlotWithTwoHistsTask(config=config)
+
+        n = len(flux)
+        self.bands, columns = config.get_requirements()
+        data = {
+            'refcat_flux': flux,
+            'patch': np.zeros(n, dtype=int),
+        }
+
+        # Assign values to columns based on their unchanged default names
+        for column in columns:
+            if column not in data:
+                if column.startswith('detect'):
+                    data[column] = np.ones(n, dtype=bool)
+                elif column.endswith('_flag') or 'Flag' in column:
+                    data[column] = np.zeros(n, dtype=bool)
+                elif column.endswith('Flux'):
+                    config.axisActions.yAction.col1 = column
+                    data[column] = flux_meas
+                elif column.endswith('FluxErr'):
+                    data[column] = flux_err
+                elif column.endswith('_extendedness'):
+                    data[column] = extendedness
+                else:
+                    raise RuntimeError(f'Unexpected column {column} in ScatterPlotWithTwoHistsTaskConfig')
+
+        self.data = pd.DataFrame(data)
+
+    def tearDown(self):
+        if os.path.exists(self.testDir):
+            shutil.rmtree(self.testDir, True)
+        del self.bands
+        del self.data
+        del self.task
+
+    def test_ScatterPlotWithTwoHistsTask(self):
```

Comment:
You also need to protect plotting tests with the default matplotlibrc:
https://github.com/LSSTDESC/skyproj/blob/702ac340e46cccae37ad91194d017816bf596ceb/tests/test_plotting.py#L17